### PR TITLE
Preliminary support for distro-independent CI jobs

### PR DESCRIPTION
**Pull request by cottsay**
This change adds support for CI jobs which are not associated with a ROS distribution. This aligns with the distro-independent documentation jobs that are already supported by `ros_buildfarm`.

### CODE REVIEW

**Review by nuclearsandwich — COMMENTED** [review]
This looks like a good start. I've not done a full review yet while the merge conflicts are outstanding. I've left one comment where I had questions/concerns and I think it would be good to add some documentation updates which will help build out the intent for this change, which I think based on prior conversation is to test the independent ROS Infrastructure packages.

I also think it would be a good idea to settle on `global` or `independent` as the way to describe these CI jobs and the symbol to use for them in the code. We have independent doc jobs so why do we have global CI jobs?

File: ros_buildfarm/ci_job.py [hunk 1]
```diff
@@ -246,9 +246,9 @@ def _get_ci_job_config(
             '%s=%s' % (var, value)
             for var, value in sorted(build_file.build_environment_variables.items())]
 
-    distribution_type = index.distributions[rosdistro_name] \
+    distribution_type = index.distributions.get(rosdistro_name, {}) \
         .get('distribution_type', 'ros1')
-    assert distribution_type in ('ros1', 'ros2')
+    assert distribution_type in (None, 'ros1', 'ros2')
```
[the remark this answers is not in the document]
Comment (cottsay) [reply ros_buildfarm/ci_job.py]: Okay, I made that change.

File: ros_buildfarm/common.py [hunk 2]
```diff
@@ -184,7 +184,7 @@ def get_ci_job_name(rosdistro_name, os_name, os_code_name, arch, job_type):
 
 
 def get_ci_view_name(rosdistro_name):
-    view_name = '%sci' % rosdistro_name[0].upper()
+    view_name = '%sci' % rosdistro_name[:1].upper()
```
Comment (nuclearsandwich) [inline ros_buildfarm/common.py:193]: This is a subtlety that might be worth a comment perhaps?

`''[0]` is an IndexError whereas `''[:1]` is `''`.

**Review by nuclearsandwich — APPROVED** [review]

File: ros_buildfarm/common.py (hunk 2 above)
Comment (cottsay) [reply ros_buildfarm/common.py:194]: Sure, 26f037d

**Pull request merged**